android/src/main/java/com/reactnativepagerview/PagerViewViewManager.kt

```diff
@@ -25,10 +26,24 @@ class PagerViewViewManager : ViewGroupManager<ViewPager2>() {
     return REACT_CLASS
   }
 
+  fun createFragmentAdapter(reactContext: ThemedReactContext): FragmentAdapter {
+    return FragmentAdapter((reactContext.currentActivity as FragmentActivity?)!!)
+  }
+
   override fun createViewInstance(reactContext: ThemedReactContext): ViewPager2 {
     val vp = ViewPager2(reactContext)
-    val adapter = FragmentAdapter((reactContext.currentActivity as FragmentActivity?)!!)
-    vp.adapter = adapter
+    if(reactContext.hasCurrentActivity()) {
+      vp.adapter = createFragmentAdapter(reactContext);
+    }
+    reactContext.addLifecycleEventListener(object : LifecycleEventListener {
```

Comment:
Please also remove the listener from reactContext instance to avoid memory leaks:
```
  override fun onDropViewInstance(view: ViewPager2) {
    super.onDropViewInstance(view)
    UIManagerHelper.getReactContext(view).removeLifecycleEventListener(this);
  }
 ```